### PR TITLE
Allow JacksonDeserializer to work with Java 9+ Map.of

### PR DESCRIPTION
**Pull request by bdemers**
and other maps that do NOT allow null keys

### CODE REVIEW

File: extensions/jackson/src/test/groovy/io/jsonwebtoken/jackson/io/JacksonDeserializerTest.groovy [hunk 1]
```diff
@@ -105,6 +105,41 @@ class JacksonDeserializerTest {
...
+                    throw new NullPointerException("key is null, expected for this test")
+                }
+                return super.containsKey(key)
+            }
+        }
+
+        // TODO: the following does NOT work with Java 1.7
+        // when we stop supporting that version we can use a partial mock instead
+        // the `typeMap.put("custom", CustomBean)` line below results in an NPE, (only on 1.7)
+
```
Comment (bdemers) [inline extensions/jackson/src/test/groovy/io/jsonwebtoken/jackson/io/JacksonDeserializerTest.groovy]: @lhazlewood any ideas on how to clean up this test?
The partial mock doesn't work on 1.7, and I'm not sure I want to dig into that, but I feel like the above solution could be better too...

Comment (lhazlewood) [reply extensions/jackson/src/test/groovy/io/jsonwebtoken/jackson/io/JacksonDeserializerTest.groovy]: I don't know about the `partialMockBuilder` myself - never used it.  That said, I like creating the anonymous subclass - that's even better than a mock IMO.

File: extensions/jackson/src/main/java/io/jsonwebtoken/jackson/io/JacksonDeserializer.java [hunk 2]
```diff
@@ -120,7 +120,8 @@ private MappedTypeDeserializer(Map<String, Class> claimTypeMap) {
         @Override
         public Object deserialize(JsonParser parser, DeserializationContext context) throws IOException {
             // check if the current claim key is mapped, if so traverse it's value
-            if (claimTypeMap != null && claimTypeMap.containsKey(parser.currentName())) {
+
+            if (claimTypeMap != null && parser.currentName() != null && claimTypeMap.containsKey(parser.currentName())) {
```
Comment (lhazlewood) [inline extensions/jackson/src/main/java/io/jsonwebtoken/jackson/io/JacksonDeserializer.java]: Should we memoize `parser.currentName()` ?  It's called in this code 3 different times.

**Comment by lhazlewood** [comment]
Also, don't forget to target a 0.11.x branch

**Comment by bdemers** [comment]
Agreed, lets get the 0.11.1 release out with this and the Gson issue out, and then cut a maintenance branch for 0.11.x so we can continue with 1.0

**Pull request merged**